packages/appium/lib/cli/extension-command.js

```diff
@@ -699,26 +717,54 @@ class ExtensionCommand {
       );
     }
 
-    const runner = new SubProcess(process.execPath, [extScripts[scriptName], ...extraArgs], {
-      cwd: this.config.getInstallPath(installSpec),
-    });
+    if (bufferOutput) {
+      const runner = new SubProcess(process.execPath, [extScripts[scriptName], ...extraArgs], {
+        cwd: this.config.getInstallPath(installSpec),
+      });
 
-    const output = new RingBuffer(50);
+      const output = new RingBuffer(50);
 
-    runner.on('stream-line', (line) => {
-      output.enqueue(line);
-      this.log.log(line);
-    });
+      runner.on('stream-line', (line) => {
+        output.enqueue(line);
+        this.log.log(line);
+      });
 
-    await runner.start(0);
+      await runner.start(0);
 
-    try {
-      await runner.join();
-      this.log.ok(`${scriptName} successfully ran`.green);
-      return {output: output.getBuff()};
-    } catch (err) {
-      this.log.error(`Encountered an error when running '${scriptName}': ${err.message}`.red);
-      return {error: err.message, output: output.getBuff()};
+      try {
+        await runner.join();
+        this.log.ok(`${scriptName} successfully ran`.green);
+        return {output: output.getBuff()};
+      } catch (err) {
+        this.log.error(`Encountered an error when running '${scriptName}': ${err.message}`.red);
+        return {error: err.message, output: output.getBuff()};
+      }
+    } else {
```

Comment:
given that we return in the block above, we don't really need the `else` and can save a level of indentation